akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -230,63 +237,46 @@ private[akka] object StreamLayout {
     override def materializedValueComputation: MaterializedValueNode = Ignore
   }
 
-  final case class CompositeModule(
-    subModules: Set[Module],
-    shape: Shape,
-    override val connections: List[(OutPort, InPort)],
-    override val materializedValueComputation: MaterializedValueNode,
-    attributes: OperationAttributes) extends Module {
+  final case class CopiedModule(shape: Shape, attributes: OperationAttributes, copyOf: Module) extends Module {
+    override val subModules: Set[Module] = Set(copyOf)
+
+    override def withAttributes(attr: OperationAttributes): Module = this.copy(attributes = attr)
+
+    override def carbonCopy: Module = this.copy(shape = shape.deepCopy())
```

Comment:
(Wrong comment I guess) So I tried that road as well, but since the stack needs an initial entry I needed to make an unnecessary copy of topLevel to make it the right type.